### ISSUE TITLE
Visual Studio Code override for package broken on MacOS

### ISSUE THREAD

**Issue by cideM** [NixOS/nixpkgs#126272]
**Describe the bug**

**TL;DR**: The commit `3a3543edd4b83489f11bff872a3b5ed0618d3b67` changed a command to use `executableName` instead of a hardcoded `code`. If you set `isInsiders = true` the `executableName` ends up being `code-insiders` but the source directory we download as an archive only includes a `code` executable on Darwin, regardless of whether you're installing insiders or not. Therefore the command that tries to operate on `executableName` fails. Reverting the commit restores the build and installation.

Only applies on MacOS, on NixOS everything works.

Ever since `3a3543edd4b83489f11bff872a3b5ed0618d3b67` I can no longer set the  `package` option of the VSCode derivation to a custom package. I am currently on `3bc8e5cd23b84b2e149e7aaad57117da16a19e6f` which is the latest commit on `nixpkgs-unstable` at the time of writing. If I run from a local `nixpkgs` checkout:

```
NIXPKGS_ALLOW_UNFREE=1 nix-env -i -E '{ ... }: let pkgs = import ./default.nix {}; in let foo = pkgs.vscode.override { isInsiders = true; }; in (pkgs.vscode-with-extensions.override { vscode = foo; }).overrideAttrs (old: { inherit (foo) pname version; })'
```

I get 

```
installing 'vscode-with-extensions-1.56.2'
this derivation will be built:
  /nix/store/5ih2wxrznzh9qjjfk6wb2i0j1q22v0b1-vscode-with-extensions-1.56.2.drv
building '/nix/store/5ih2wxrznzh9qjjfk6wb2i0j1q22v0b1-vscode-with-extensions-1.56.2.drv'...


Builder called die: Cannot wrap '/nix/store/xjr0qjgxpv8l125sdh20iypqmdpa1ix5-vscode-1.56.2/bin/code-insiders' because it is not an executable file
Builder called die: Cannot wrap '/nix/store/xjr0qjgxpv8l125sdh20iypqmdpa1ix5-vscode-1.56.2/bin/code-insiders' because it is not an executable file
Backtrace:
Backtrace:
7 assertExecutable /nix/store/zj4aacsc6gjzchgcnipwgisfmb4bh4wf-hook/nix-support/setup-hook
7 assertExecutable /nix/store/zj4aacsc6gjzchgcnipwgisfmb4bh4wf-hook/nix-support/setup-hook
36 makeWrapper /nix/store/zj4aacsc6gjzchgcnipwgisfmb4bh4wf-hook/nix-support/setup-hook
36 makeWrapper /nix/store/zj4aacsc6gjzchgcnipwgisfmb4bh4wf-hook/nix-support/setup-hook
8 source /private/tmp/nix-build-vscode-with-extensions-1.56.2.drv-0/.attr-0
8 source /private/tmp/nix-build-vscode-with-extensions-1.56.2.drv-0/.attr-0
1281 genericBuild /nix/store/vnazyva3nawmvmhsxl8d2698wqg3743l-stdenv-darwin/setup
1281 genericBuild /nix/store/vnazyva3nawmvmhsxl8d2698wqg3743l-stdenv-darwin/setup
2 main /nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh
2 main /nix/store/9krlzvny65gdc8s7kpb6lkx8cd02c25b-default-builder.sh


builder for '/nix/store/5ih2wxrznzh9qjjfk6wb2i0j1q22v0b1-vscode-with-extensions-1.56.2.drv' failed with exit code 1
error: build of '/nix/store/5ih2wxrznzh9qjjfk6wb2i0j1q22v0b1-vscode-with-extensions-1.56.2.drv' failed
```

if I `git revert 3a3543edd4b83489f11bff872a3b5ed0618d3b67` the above command works just fine.

Furthermore, if I add `ls "$out/Applications/${longName}.app/Contents/Resources/app/bin/"` right before the `ln -s` call I only see a `code` executable, whereas `${executableName}` ends up being `code-insiders` for me. In other words, the commit I need to revert makes sense actually, so the actual bug seems to be that there's only a `code` file. But at this point I need to sleep. I don't know if it's us in Nixpkgs that decide the `code` name or if this comes from the source repo.

@jonringer
@thirstman 

**UPDATE:** I opened the source that was mentioned in the builders log 

```
~/nixpkgs on nixpkgs-unstable •
❯ NIXPKGS_ALLOW_UNFREE=1 nix-env -i -E '{ ... }: let pkgs = import ./default.nix {}; in let foo = pkgs.vscode.override { isInsiders = true; }; in (pkgs.vscode-with-extensions.override { vscode = foo; }).overrideAttrs (old: { inherit (foo) pname version; })'
replacing old 'vscode-with-extensions-1.56.2'
installing 'vscode-with-extensions-1.56.2'
querying info about '/nix/store/85ajdxm5g85d4yqcnfilq55lx0fx57hn-vscode-with-extensions-1.56.2' on 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/85ajdxm5g85d4yqcnfilq55lx0fx57hn.narinfo'...
querying info about '/nix/store/bdjh10a26rlg4fs4wbgi7apqa77qzi77-vscode-1.56.2' on 'https://cache.nixos.org'...
downloading 'https://cache.nixos.org/bdjh10a26rlg4fs4wbgi7apqa77qzi77.narinfo'...
these 2 derivations will be built:
  /nix/store/d1wia3yqjkxmzdicvgk96zr902ic216r-vscode-1.56.2.drv
  /nix/store/l473r91dqfas73kk8lb5bks2jsvixdgy-vscode-with-extensions-1.56.2.drv
building '/nix/store/d1wia3yqjkxmzdicvgk96zr902ic216r-vscode-1.56.2.drv'...
unpacking sources
unpacking sources
unpacking source archive /nix/store/67sf2p7y6lshj2bprvia3z3pzsjkd810-VSCode_1.56.2_darwin.zip <-- THIS FILE
unpacking source archive /nix/store/67sf2p7y6lshj2bprvia3z3pzsjkd810-VSCode_1.56.2_darwin.zip
source root is Visual Studio Code.app
source root is Visual Studio Code.app
```

and indeed
```
~/Downloads/Visual Studio Code.app/Contents
$ ls Resources/app/bin/
code
```

Only code, no code-insiders. If I go to https://code.visualstudio.com/insiders/ and download the "Universal" installer (?) and peek inside the `.app` I also just see `code`. If I download the insiders installer for Linux as `.tar.gz` it does contain a `code-insiders` executable!

I don't really understand how this commit worked for the author though since they intentionally modified the Darwin code path so I assume this worked for them? 🤔 But since I'm doing all of this with the linked Nix expression from a local checkout it shouldn't really matter what is installed on my system.

**Comment by jonringer** [comment]
seems like we need more than just `executableName` to describe the desired command in output, but something more like `originalName` and `installedName`.

That way we can have `code`, `codium`, `code-insiders`, and `codium-insiders`.

**Comment by cideM** [comment]
But what is the purpose of replacing `code` with a variable when `code` is not in fact variable? The archive appears to always just include a file called `code`.

**Comment by minhtrancccp** [comment]
> VSCodium Insiders doesn't exist [VSCodium/vscodium#148](https://github.com/VSCodium/vscodium/issues/148)

as https://github.com/VSCodium/vscodium/issues/148 has been resolved since, will support for [`vscodium-insiders`](https://github.com/VSCodium/vscodium-insiders) be provided now? many thanks.
